addons/stock/models/stock_picking.py

```diff
@@ -265,6 +265,10 @@ class Picking(models.Model):
         default=lambda self: self.env['res.company']._company_default_get('stock.picking'),
         index=True, required=True,
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]})
+    user_id = fields.Many2one(
+        'res.users', 'Responsible', tracking=True,
+        domain=lambda self: [('groups_id', 'in', self.env.ref('stock.group_stock_user').id)],
+        default=lambda self: self.env.user)
```

Comment:
@jip-odoo what if the picking is created by a reording rule ? who will be the responsible ?

Reply:
i have updated changes as per your comment
if picking is create automatically in this case no one will be the responsible